module/zfs/spa_misc.c

```diff
@@ -1853,7 +1853,8 @@ spa_update_dspace(spa_t *spa)
 		spa_config_enter(spa, SCL_VDEV, FTAG, RW_READER);
 		vdev_t *vd =
 		    vdev_lookup_top(spa, spa->spa_vdev_removal->svr_vdev_id);
-		if (vd->vdev_mg->mg_class == spa_normal_class(spa)) {
+		if (vd->vdev_mg &&
```

Comment:
This makes sense.  After `vdev_remove_complete()` clears `vdev_mg` (and also removes it from the accounting used by `metaslab_class_get_dspace(spa_normal_class)`, it drops the spa_config lock while waiting for the `vdev_remove_complete_sync()` sync task to run, which will clear `spa_vdev_removal`.  So we could be catching it in between, when there's still a `spa_vdev_removal` but it doesn't have a `vdev_mg`.

Could you add a comment here explaining this?

Reply:
Sure, I'll give it a whirl.